packages/frontend-v2/hooks/utils.ts

```diff
@@ -0,0 +1,15 @@
+import {AxiosError} from "axios";
+import {useRouter} from "next/router";
+
+export const useRedirectUnAuth = () => {
+  const router = useRouter();
+  return (error: AxiosError) => {
+    if (
+      error.response &&
+      (error.response.status === 401 || error.response.status === 403)
```

Comment:
why redirect on 401 or 403? i think it would make more sense to instead check if user is authorized on page load, and if not redirect to login page (something like [this](https://github.com/HackBeanpot/application-portal/blob/9f2ec318cca451d828b8ca651a7e8980f5b5ed19/hooks/useSessionOrRedirect.ts))

Reply:
ah yea 403 should not redirect to sign-in since that should mean that the user is authenticated just not authorized.

Also, I had suggested this but i think delegating the work of checking whether the user is authorized or not to the component before accessing protected api routes makes sense.

We can remove checks for 401s when accessing protected data in hooks and instead return just return the error. Components can be responsible for ensuring the user is authenticated. If the user is authenticated and we get a 403 then we components can display an appropriate message.